src/transformers/training_args.py

```diff
@@ -1368,16 +1375,41 @@ def _setup_devices(self) -> "torch.device":
             device = torch.device("cuda", self.local_rank)
             self._n_gpu = 1
         elif self.local_rank == -1:
-            # if n_gpu is > 1 we'll use nn.DataParallel.
-            # If you only want to use a specific subset of GPUs use `CUDA_VISIBLE_DEVICES=0`
-            # Explicitly set CUDA to the first (index 0) CUDA device, otherwise `set_device` will
-            # trigger an error that a device index is missing. Index 0 takes into account the
-            # GPUs available in the environment, so `CUDA_VISIBLE_DEVICES=1,2` with `cuda:0`
-            # will use the first GPU in that env, i.e. GPU#1
-            device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-            # Sometimes the line in the postinit has not been run before we end up here, so just checking we're not at
-            # the default value.
-            self._n_gpu = torch.cuda.device_count()
+            if self.use_mps_device:
```

Comment:
Should this also be checked under the `no_cuda` conditional block in the beginning of the method? In case one selects the arguments `no_cuda` and `use_mps_device`.

Reply:
The purpose of `no_cuda` is `to not use CUDA even when it is available`. When using an Apple silicon device, there are no Nvidia cuda based devices and users will never use this option. Hence, I think this is not required.